models/velodrome/optimism/velodrome_optimism_trades.sql

```diff
@@ -52,14 +81,14 @@ SELECT
         when lower(erc20a.symbol) > lower(erc20b.symbol) then concat(erc20b.symbol, '-', erc20a.symbol)
         else concat(erc20a.symbol, '-', erc20b.symbol)
     end as token_pair
-    ,dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount
-    ,dexs.token_sold_amount_raw / power(10, erc20b.decimals) AS token_sold_amount
-    ,dexs.token_bought_amount_raw  AS token_bought_amount_raw
-    ,dexs.token_sold_amount_raw  AS token_sold_amount_raw
+    ,cast(dexs.token_bought_amount_raw as double) / cast( power(10, erc20a.decimals) as double) AS token_bought_amount
+    ,cast(dexs.token_sold_amount_raw as double) / cast( power(10, erc20b.decimals) as double) AS token_sold_amount
```

Comment:
these likely don't need casted either

Reply:
If we don't cast, I think we'll lose the decimals (basing on queries in Dune UI, I always have to cast as double)

Reply:
we don't do it on other dex models, i.e. here:
https://github.com/duneanalytics/spellbook/blob/main/models/uniswap/ethereum/uniswap_v3_ethereum_trades.sql#L56-L57

when you merge into `dex.trades`, it may fail on the union on data type potentially.

we have a beta `dex.trades` until migration is final (only two projects away), you should try adding it in this PR to include this project into sector spell and see if union works fine
https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_trades_migration_beta.sql

Reply:
You'll only perform integer division (which loses the decimals) if both sides are integer types. 
`pow(x,y)` is double by default so removing the casts shouldn't be a problem. :ok_hand: 
https://dune.com/queries/2982847